crates/astria-composer/src/ds/eth_provider.rs

```diff
@@ -0,0 +1,92 @@
+use std::time::Duration;
+
+use async_trait::async_trait;
+use humantime::format_duration;
+use tracing::{debug, info, instrument, warn};
+use color_eyre::eyre::{self, Context};
+use tokio::sync::mpsc as tokio_mpsc;
+
+use ethers::providers::{
+    Middleware, Provider as EthersProvider, ProviderError as EthersProviderError, StreamExt, Ws,
```

Comment:
why do we need the type aliases here, instead of simply `Provider` and `ProviderError`? this module only implements our provider for eth